src/Mvc/src/Microsoft.AspNetCore.Mvc.RazorPages/CompiledPageActionDescriptor.cs

```diff
@@ -59,5 +60,10 @@ public CompiledPageActionDescriptor(PageActionDescriptor actionDescriptor)
         /// Gets or sets the <see cref="TypeInfo"/> of the page.
         /// </summary>
         public TypeInfo PageTypeInfo { get; set; }
+
+        /// <summary>
+        /// Gets or sets the associated <see cref="Endpoint"/> of this page.
+        /// </summary>
+        public Endpoint Endpoint { get; set; }
```

Comment:
Is it possible that the same compiled page could be referenced by multiple route endpoints? I am concern about what happens if they are setup with different metadata and how that would effect the endpoint here.

Or is it not a problem because different `CompiledPageActionDescriptor` instances are used for each?

Reply:
No, that is not possible. Pages always use attribute routing, so if you have two routes you always have to action descriptors.

Conventional routes + controllers allow the same ad to be used multiple times. 

A good question!